src/pages/tutorials/build-your-first-app/setting-up-app.md

```diff
@@ -0,0 +1,660 @@
+
+1. Create a file named `src/utils.ts` and add the following function to retrieve the exchange rates:
+
+   ```typescript
+   export const fetchExchangeRates = async (): Promise<{
+     tzbtcPrice: number;
+     xtzPrice: number;
+   } | null> => {
+     const query = `
+         query {
+           overview { xtzUsdQuote },
+           token(id: "KT1PWx2mnDueood7fEmfbBDKx1D9BAnnXitn") { price }
+         }
+       `;
+     const res = await fetch(`https://analytics-api.quipuswap.com/graphql`, {
+       method: "POST",
+       headers: {
+         "Content-Type": "application/json"
+       },
+       body: JSON.stringify({
+         query
+       })
+     });
+     if (res.status === 200) {
+       const resData = await res.json();
+       let xtzPrice = resData?.data?.overview?.xtzUsdQuote;
+       let tzbtcPrice = resData?.data?.token?.price;
+       // validates the 2 values
+       if (xtzPrice && tzbtcPrice) {
+         xtzPrice = +xtzPrice;
+         tzbtcPrice = +tzbtcPrice;
+         if (!isNaN(xtzPrice) && !isNaN(tzbtcPrice)) {
+           // tzBTC price is given in XTZ by the API
+           tzbtcPrice = tzbtcPrice * xtzPrice;
+           return { tzbtcPrice, xtzPrice };
+         }
+       } else {
+         return null;
+       }
+     } else {
+       return null;
+     }
+   };
+   ```
+
+   This function uses the [QuipuSwap GraphQL API](https://analytics-api.quipuswap.com/graphql) to fetch the exchange rates.
+   Then the function parses them and validates the prices for XTZ and tzBTC.
+
+1. Add this code to the top of the `utils.ts` file, which includes functions to format token amounts:
+
+   ```typescript
+   import BigNumber from "bignumber.js";
+   import type { TezosToolkit } from "@taquito/taquito";
+   import type { token } from "./types";
+   import type { TezosAccountAddress } from "./store";
+   import { tzbtcAddress, sirsAddress } from "./config";
+
+   export const displayTokenAmount = (
+     amount_: BigNumber | number,
+     token: token
+   ): string => {
+     let amount = BigNumber.isBigNumber(amount_) ? amount_.toNumber() : amount_;
+     switch (token) {
+       case "XTZ":
+         return (+(amount / 10 ** 6).toFixed(6)).toLocaleString("en-US");
+       case "tzBTC":
+         if (amount < 100) {
+           return "> 0.000001";
+         }
+         return (+(amount / 10 ** 8).toFixed(8)).toLocaleString("en-US", {
+           maximumSignificantDigits: 8
+         });
+       case "SIRS":
+         return (+amount.toFixed(2)).toLocaleString("en-US");
+     }
+   };
+
+   export const shortenHash = (hash: string): string =>
+     hash ? hash.slice(0, 5) + "..." + hash.slice(-5) : "";
+
+   export const calcDeadline = () =>
+   new Date(Date.now() + 3_600_000).toISOString();
+   ```
+
+1. Copy these files from the tutorial repository at <https://github.com/trilitech/tutorial-applications/tree/main/liquidity-baking-dapp>, making sure to put the files in the sale folder in your application.
```

Comment:
This repo is private? we could write a simple shell script to get the files created and populated with the right content from github repo (this example is not working as the repo is private, but still would save a lot of time: 
```# Create directories
mkdir -p src/lib

# Download and save content from GitHub
curl -o src/config.ts https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/config.ts
curl -o src/store.ts https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/store.ts
curl -o src/types.ts https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/types.ts
curl -o src/lib/Interface.svelte https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/lib/Interface.svelte
curl -o src/lib/Sidebar.svelte https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/lib/Sidebar.svelte
curl -o src/lib/Toast.svelte https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/lib/Toast.svelte
curl -o src/lib/UserStats.svelte https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/lib/UserStats.svelte

```

Reply:
I have made the repo public.

Reply:
nice! So the command works now properly and that will save some time, there's another place we can use this way of copying the content, I'll point it out in a separate comment.

Reply:
Not sure about the shell script. If the user is not experienced with them, copying that large script might be more trouble than downloading the files manually.

Reply:
Fair enough - but would be good to provide URL for each raw file next to it eg.: `src/config.ts` -> `https://raw.githubusercontent.com/trilitech/tutorial-applications/main/liquidity-baking-dapp/src/config.ts`